### PR TITLE
Remove unnecessary version numbers from docs.

### LINKED ISSUE #2422: Remove unnecessary version numbers from docs
See for example : 

CleanShot 2023-08-02 at 10 45 21

### PR DESCRIPTION
**Pull request by wetted**
closes #2422

### CODE REVIEW

**Review by sdelamo — CHANGES_REQUESTED** [review]
I don't think we can remove the Jakarta Javax persistence versions since they are not managed. 

https://github.com/micronaut-projects/micronaut-data/blob/master/gradle/libs.versions.toml#L25-L28

Same for spring-orm 

@radovanradic @dstepanov is there a reason why those persistence versions are not managed?

File: src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc [hunk 1]
```diff
@@ -71,11 +71,11 @@ NOTE: The `schema-generate` setting is only useful for demos and testing trivial
 
 To retrieve objects from the database you need to define a class annotated with ann:data.annotation.MappedEntity[]. Note that this is a meta annotation and in fact if you prefer you can use JPA annotations (only a subset are supported, more on that later). If you wish to use JPA annotations include the following `compileOnly` scoped dependency:
 
-dependency:jakarta.persistence:jakarta.persistence-api[version="3.0.0", scope="compileOnly"]
+dependency:jakarta.persistence:jakarta.persistence-api[scope="compileOnly"]
 
 To use JPA annotations in the `javax.persistence` package use:
 
-dependency:jakarta.persistence:jakarta.persistence-api[version="2.2.3", scope="compileOnly"]
+dependency:jakarta.persistence:jakarta.persistence-api[version="3.0.0", scope="compileOnly"]
```
Comment (sdelamo) [inline src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc:78]:
```suggestion
dependency:javax.persistence:javax.persistence-api[scope="compileOnly"]
```

**Review by sdelamo — APPROVED** [review]

File: src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc (hunk 1 above)
Comment (sdelamo) [reply src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc:78]: @radovanradic @dstepanov I assume this was a bug we were not mentioning the correct maven coordinates. 

[Making the dependencies managed](https://github.com/micronaut-projects/micronaut-data/pull/2428), should allow us to remove the version number.

File: src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc [hunk 2]
```diff
@@ -71,11 +71,11 @@ NOTE: The `schema-generate` setting is only useful for demos and testing trivial
 
 To retrieve objects from the database you need to define a class annotated with ann:data.annotation.MappedEntity[]. Note that this is a meta annotation and in fact if you prefer you can use JPA annotations (only a subset are supported, more on that later). If you wish to use JPA annotations include the following `compileOnly` scoped dependency:
 
-dependency:jakarta.persistence:jakarta.persistence-api[version="3.0.0", scope="compileOnly"]
+dependency:jakarta.persistence:jakarta.persistence-api[scope="compileOnly"]
```
Comment (sdelamo) [inline src/main/docs/guide/dbc/jdbc/jdbcQuickStart.adoc:74]: [Making the dependencies managed](https://github.com/micronaut-projects/micronaut-data/pull/2428), should allow us to remove the version number.

**Review by radovanradic — APPROVED** [review]

**Comment by sdelamo** [comment]
> @sdelamo I'm a little confused. Should I change this PR to change the `jakarta/javax-persistence` and `spring-orm` to managed dependencies on this PR, or should we create a separate issue for it.

I created a separate PR: https://github.com/micronaut-projects/micronaut-data/pull/2428

I am not sure about spring-orm as a managed dependency.

**Pull request merged**